File: 1. Two Sum/1. Two Sum.php
class Solution {
    /**
     * @param Integer[] $nums
     * @param Integer $target
     * @return Integer[]
     */
    function twoSum($nums, $target) {
        $numMap = array();
        
        foreach ($nums as $i => $num) {
            $complement = $target - $num;
            if (isset($numMap[$complement])) {
                return array($numMap[$complement], $i);
            }
            $numMap[$num] = $i;
        }
        
        // This should not happen given the problem constraints
        return array();
    }
}

File: 1. Two Sum/1. Two Sum.swift
class Solution {
    func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
        var numMap = [Int: Int]()
        
        for (i, num) in nums.enumerated() {
            let complement = target - num
            if let complementIndex = numMap[complement] {
                return [complementIndex, i]
            }
            numMap[num] = i
        }
        
        // This should not happen given the problem constraints
        return []
    }
}